### ISSUE TITLE
Improve email compatibility for WebP images

### ISSUE THREAD

**Issue by scofennell** [adamsilverstein/modern-images-wp#25]
Unfortunately, the webp image format is not well supported in some versions of Outlook:

https://www.caniemail.com/features/image-webp/

Although it's not a large % globally, I run a fairly run-of-the-mill platform for law firms, and for our audience the number is more like 15%.

In our case, our emails come from MailChimp scraping our RSS feeds, and so it would be nice to have a jpeg version available for use in RSS (assuming the blog author has uploaded a jpeg that we are converting to webp).  And it would be nice if it were not the "full" image file, which can be way, way too large.

I'm not sure of the best way to go about this.

Any thoughts?  Plans?  Suggestions?

**Comment by adamsilverstein** [comment]
Hi @scofennell - thanks for raising this issue. Good point about mail clients not supporting WebP, I hadn't thought of images in that context.

Some thoughts:

* we are working on supporting "multiple image formats" in https://github.com/adamsilverstein/modern-images-wp/issues/1, this approach would use the `<picture>` element which enables supporting multiple formats, letting browser "fall back" top a type they support. The original intent was to support AVIF and JPEGXL which have limited browser support, but the same approach could be used for WebP. Still I'm not certain this approach works for Outlook, we would have to test that!

* Consider opening a ticket with Mail Chimp where you are a customer - since a part of what their service does is prepare your content for emailing and the various idiosyncratic email clients out there. Given the increasing adoption of WebP on the web, they may (or maybe should) have plans to do their own WebP handling. Although given the fact that they would have to re-compress the lossy WebP files, this isn't really an ideal solution - image quality will sufer.

* Final option once https://github.com/adamsilverstein/modern-images-wp/issues/1 is in place: could could use WordPress filters to set image format for the RSS feed to JPEG, while keeping WebP for the front end/main site. This capability - to use specific image types for RSS - might also be appropriate for core, if we determine that mail client (or other RSS consumers) WebP support is poor.

**Comment by scofennell** [comment]
Yeah it seems like point number three is fairly promising.

<picture> support in email is quite poor:  https://www.caniemail.com/search/?s=picture

**Comment by adamsilverstein** [comment]
Very interesting, I hadn't seen `caniemail` before, thanks for sharing those links Looks like WebP support is pretty good fortunately, only outlook on windows completely lacks support. Does this apply to **all** versions of Windows? I'm a little surprised support isn't contingent on the version of the underlying operating system.

**Comment by scofennell** [comment]
That's a good question and I don't quite understand it myself.  Here's where I'm coming from:

In MailChimp, they offer an integration with the excellent "Litmus" service, for getting screenshots from various email clients.  I'm getting the webp's NOT rendering in a variety of Outlook versions, and it totals to approx 15% of my stodgy, corporate audience.

It's unclear to me how those versions correspond to the caniemail numbers.

Regardless, it seems like the roadmap here is to leverage some ability to convert to multiple formats.

**Comment by adamsilverstein** [comment]
@scofennell thanks for the additional details and sharing your real-world data. Will definitely continue working with this consideration in mind.
